### PR TITLE
Add docker support

### PR DESCRIPTION
**Pull request by benjick**
Example usage with docker-compose

``` yml
nzedb:
  build: .
  restart: always
  links:
    - db:db
  ports:
    - "3000:80"

db:
  image: mariadb
  restart: always
  environment:
    MYSQL_PASSWORD: superSecretRootPassword
    MYSQL_ROOT_PASSWORD: myDbPassword
    MYSQL_USER: nzedb
    MYSQL_DATABASE: nzedb

# docker-compose up
```

Database hostname would be `db`

### CODE REVIEW

**Comment by niel** [comment]
It would be better to put the extra files in .../build/docker instead of under misc.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+COPY ./misc/docker/nginx /etc/nginx/sites-available/nZEDb
+COPY ./misc/docker/init.sh /init.sh
+
+RUN apt-get install -y software-properties-common && \
+  add-apt-repository ppa:ondrej/php5-5.6 && \
+  add-apt-repository ppa:nginx/stable && \
+  apt-get update && \
+  apt-get install -y --force-yes \
+    software-properties-common \
```
Comment (jangrewe) [inline Dockerfile]: duplicate

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  && unlink /etc/nginx/sites-enabled/default \
+  && ln -s /etc/nginx/sites-available/nZEDb /etc/nginx/sites-enabled/nZEDb \
+  && git clone https://github.com/nZEDb/nZEDb.git /var/www/nZEDb \
+  && chmod -R 777 /var/lib/php5/sessions \
+  && chmod -R 777 /var/www/nZEDb/libs/smarty/templates_c \
+  && chmod -R 777 /var/www/nZEDb/resources \
+  && mkdir -p /var/www/nZEDb/nzedb/config \
+  && chmod -R 777 /var/www/nZEDb/nzedb/config \
+  && chmod -R 777 /var/www/nZEDb/www \
+  && apt-get clean \
```
Comment (jangrewe) [inline Dockerfile]: 'apt-get clean' is unnecessary (i did that a lot, too)

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,44 @@
+FROM ubuntu:14.04
+
+COPY ./misc/docker/nginx /etc/nginx/sites-available/nZEDb
+COPY ./misc/docker/init.sh /init.sh
+
+RUN apt-get install -y software-properties-common && \
+  add-apt-repository ppa:ondrej/php5-5.6 && \
+  add-apt-repository ppa:nginx/stable && \
+  apt-get update && \
+  apt-get install -y --force-yes \
```
Comment (jangrewe) [inline Dockerfile]: --force-yes may not be needed, but --no-install-recommends helps to keep the image size down

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,44 @@
+FROM ubuntu:14.04
+
```
Comment (jangrewe) [inline Dockerfile]: add "ENV DEBIAN_FRONTEND noninteractive" to supresses loads of warnings, less clutter

**Comment by zetas** [comment]
FWIW i tried running this and never got it to work. I use nzedb-docker made by razorgirl but would much rather use the newer docker-compose instead. I ran the commands suggested using the compose and docker file and was unable to stand up a new install.

**Comment by ggee** [comment]
Can this be updated to the latest installation method using the new compose for the 0.x branch?

**Comment by slydetector** [comment]
@ggee Working with docker at https://github.com/slydetector/simply-nzedb

**Comment by niel** [comment]
No idea what happened here, but I did NOT close this and it will not let me reopen it.

**Comment by niel** [comment]
Ahh, seems the repo that the PR came from was deleted.

**Pull request closed without merging**